### PR TITLE
Make ES files inside Docker container world readable

### PR DESCRIPTION
**Pull request by pugnascotia**
Running the Elasticsearch Docker image with a different GID is possible but trappy, since at present all the ES files are only readable by the user and group. This PR documents a Docker CLI flag that fixes this situation, by ensuring the container user is added to the default group (which is `root`, GID 0).

I also added a test for this case, and refactored the Docker tests to use a builder pattern for constructing the `docker run` command. The existing code was becoming unwieldy and hard to change.

### CODE REVIEW

**Review by rjernst — APPROVED** [review]
LGTM, though my docker knowledge is limited, so having some else look over this would be good

**Review by joegallo — APPROVED** [review]

File: docs/reference/setup/install/docker.asciidoc [hunk 1]
```diff
@@ -234,6 +235,12 @@ chmod g+rwx esdatadir
 chgrp 0 esdatadir
 --------------------------------------------
 
+You can also run an {es} container using both a custom UID and GID. Unless you
+bind-mount each of the `config`, data` and `logs` directories, you must pass
+the command line option `--group-add 0` to `docker run`. This ensures that the user
```
Comment (mark-vieira) [inline docs/reference/setup/install/docker.asciidoc:240]: I guess my only question is what is the point of this? I only assume running as a different user is something done by security conscious orgs that don't like the idea of running containers as root. Wouldn't then subsequently adding the user to the root group defeat the purpose?

**Review by mark-vieira — APPROVED** [review]
One question regarding the use case here, otherwise LGTM

File: docs/reference/setup/install/docker.asciidoc (hunk 1 above)
Comment (pugnascotia) [reply docs/reference/setup/install/docker.asciidoc:240]: It's mostly a convenience for users who might be experimenting with the image, or are only interested in capturing the stdout/stderr output of the container and don't want to bind-mount the `logs` directory. It also serves to document why someone might have problems running the image with a custom GID.

Note that the container user is only added to the `root` group inside the container, but files in bind-mounted volumes are still written as the user-specified GID, so everything is still OK outside of the container.

**Pull request merged**